### DISCUSSION TITLE
Remove default focus ring opacity?

### DISCUSSION THREAD

**Discussion by Libertie** [tailwindlabs/tailwindcss#10662] [Help] [answered]
I am overriding the default ring color in tailwind.config.js with a custom color:
```
ringColor: {
    DEFAULT: 'rgb(236 72 153 / 1.0)', // pink-500
},
```
But elements with `focus:ring-2 outline-none` are getting a ring with 50% opacity. What is the best way to remove the 50% transparency as a default? Any advice appreciated!

**Comment by wongjn** [comment, marked as answer]
Two ways:

## Set a `future` feature flag:

```diff
 /** @type {import('tailwindcss').Config} */
 module.exports = {
+  future: {
+    respectDefaultRingColorOpacity: true,
+  },
   theme: {
     extend: {
       ringColor: {
         DEFAULT: 'rgb(236 72 153 / 1.0)', // pink-500
       },
     },
   },
 }
```
[Tailwind Play](https://play.tailwindcss.com/K0x5DHldgU)

## Set `ringOpacity.DEFAULT`

```diff
 /** @type {import('tailwindcss').Config} */
 module.exports = {
   theme: {
     extend: {
+      ringOpacity: {
+        DEFAULT: '1',
+      },
       ringColor: {
         DEFAULT: 'rgb(236 72 153 / 1.0)', // pink-500
       },
     },
   },
 }
```
[Tailwind Play](https://play.tailwindcss.com/Nx2O6jL6FR?file=config)

**Reply by Libertie** [reply]
Thank you! Love having options 😁
